src/wallet/rpcwallet.cpp

```diff
@@ -236,6 +236,128 @@ static void SetFeeEstimateMode(const CWallet& wallet, CCoinControl& cc, const Un
     }
 }
 
+static RPCHelpMan listaddresses()
+{
+    return RPCHelpMan{"listaddresses",
+                "\nLists wallet addresses\n",
+                {
+                    {"start_index", RPCArg::Type::NUM, RPCArg::Default{0}, "Address list starting index."},
+                    {"end_index", RPCArg::Type::NUM, RPCArg::Default{10}, "Address list ending index."},
+                },
+                RPCResult{
+                    RPCResult::Type::OBJ, "", "",{
+                    {
+                        RPCResult::Type::ARR, "", "",
+                        {
+                            {RPCResult::Type::OBJ, "", "", {
+                                {RPCResult::Type::NUM, "index", "The address position"},
+                                {RPCResult::Type::STR, "address", "The bitcoin address"},
+                                {RPCResult::Type::STR, "hdkeypath", "The BIP 32 HD path of the address"},
+                                {RPCResult::Type::STR, "descriptor", "The address descriptor"},
+                                {RPCResult::Type::BOOL, "internal", "The address is internal (change) or external (receive)"},
+                                {RPCResult::Type::NUM, "amount", "Available amount at address"},
```

Comment:
Addresses don't hold funds. They only receive, never send.

Instead, it should just tally the total received ever.

Reply:
The code to count the total received is already available in the function. It can be done.

But address explorers in most software wallets, such as Electrum and Specter Desktop, only show the funds available to be spent at each address. I think the idea is to see how the UTXO values are distributed across the addresses (if there is any significant gap, for example).